### ISSUE TITLE
Creating an App with xip.io and then upgrading the app switching to Hostname allows to leave the Hostname blank.

### ISSUE THREAD

**Issue by izaac** [rancher/rancher#22733]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug.

**Steps to reproduce (least amount of steps as possible):**
- Launch a Grafana App
  - In the Load Balancing section Enable Global DNS and enter an FQDN for it
  - Keep the **Generate automatic hostname with `xip.io` selected.** 
- Fill all required fields and hit Launch
- Upgrade the same App.
  - Select `Specify a hostname to use` this will disable `xip.io`
  - Leave the Hostname field blank.
- Click **Upgrade**

**Result:**
The app allows the user to leave the Hostname Field blank if the users return to make any upgrades to the App configuration they will end with this issue:

https://github.com/rancher/rancher/issues/22731

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.3.0-rc1
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Imported (HA) 
- Machine type (cloud/VM/metal) and specifications (CPU/memory): cloud

**Comment by loganhz** [comment]
I think we need to make it as a required field in question.yaml

**Comment by guangbochen** [comment]
fixed in https://github.com/rancher/charts/pull/223, can we please use `k8s-1.16` branch to test it, thanks.

**Comment by deniseschannon** [comment]
k8s-1.16 has been merged to master as of v2.3.2 so it can be validated on master

**Comment by soumyalj** [comment]
Tested with v2.3.2 and used Grafana version 3.8.6
Steps:
1. Launch grafana multicluster app and choose "Automatically generate a .xip.io hostname"
2. After the app is launched successfully, upgrade the app and choose "Specify a hostname"
and leave the field blank. The app allows the user to leave the Hostname Field blank and upgrade goes through fine.

If we upgrade again, and choose  "Automatically generate a .xip.io hostname", then we see the below error
![Screen Shot 2019-11-08 at 11 48 25 AM](https://user-images.githubusercontent.com/18536626/68506187-e2c3a080-021d-11ea-8bef-d40044a0b7df.png)

**Comment by loganhz** [comment]
@soumyalj the original issue is a chart issue which has been fixed. Please verify it in app. The issue you mentioned is a frontend issue in multiple cluster apps. Could you please file a new issue and assign it to UI team, please? It looks like all required fields will not be validated when you upgrade a multiple cluster app.

**Comment by loganhz** [comment]
Before we can reproduce this issue in project level app as well. So we changed the question.yaml to fix it in Chart side.

However, there is still another UI issue in multiple cluster apps. It looks like all required fields will not be validated when you upgrade a multiple cluster app.

So we still need a fix in multiple cluster app side

**Comment by westlywright** [comment]
Available in `latest-2.3`.

**Comment by soumyalj** [comment]
Tested with v2.3-head using UI index latest-2.3 . ssue 1:
Steps
Launch grafana multicluster app and choose "Automatically generate a .xip.io hostname"
After the app is launched successfully, upgrade the app and choose "Specify a hostname"
and leave the field blank. The app gives the error as expected. This issue is fixed.

<img width="790" alt="Screen Shot 2019-11-13 at 11 15 19 AM" src="https://user-images.githubusercontent.com/18536626/68795997-e6349e80-0606-11ea-818a-4d0c36d9e096.png">

A different issue observed is as below. 
Once the "Hostname is required" error is seen, click on cancel . 
Click on "upgrade" again , then "Specify hostname" is selected as below:
<img width="855" alt="Screen Shot 2019-11-13 at 11 01 47 AM" src="https://user-images.githubusercontent.com/18536626/68796150-33187500-0607-11ea-9135-ff5dd97ddc5a.png">

**Comment by westlywright** [comment]
Available in `latest-2.3`.

**Comment by soumyalj** [comment]
Steps
Launch grafana multicluster app and choose "Automatically generate a .xip.io hostname"
After the app is launched successfully, upgrade the app and choose "Specify a hostname"
and leave the field blank. The app gives the error as expected. This issue is fixed.
Once the "Hostname is required" error is seen, click on cancel .
Click on "upgrade" again , then "Automatically generate a .xip.io hostname" is selected as expected
<img width="1729" alt="Screen Shot 2019-11-13 at 2 43 50 PM" src="https://user-images.githubusercontent.com/18536626/68811038-82b96980-0624-11ea-8f9e-662ac3f2dc0d.png">
